src/components/FilePreview/FilePreview.tsx

```diff
@@ -0,0 +1,129 @@
+import React, {useEffect, useState} from 'react';
+import type {FC, MouseEventHandler} from 'react';
+
+import {
+    FileZipper as ArchiveIcon,
+    Code as CodeIcon,
+    FileQuestion as DefaultIcon,
+    Picture as ImageIcon,
+    MusicNote as MusicIcon,
+    LogoAcrobat as PdfIcon,
+    LayoutHeaderCellsLarge as TableIcon,
+    TextAlignLeft as TextIcon,
+    Filmstrip as VideoIcon,
+} from '@gravity-ui/icons';
+import {Card, Icon, IconData, Text, Tooltip, useUniqId} from '@gravity-ui/uikit';
+
+import {block} from '../utils/cn';
+
+import {FilePreviewAction, FilePreviewActionProps} from './FilePreviewAction';
+import {FileType} from './types';
+import {getFileType} from './utils';
+
+import './FilePreview.scss';
+
+const cn = block('file-preview');
+
+const FILE_ICON: Record<FileType, IconData> = {
+    default: DefaultIcon,
+    image: ImageIcon,
+    video: VideoIcon,
+    code: CodeIcon,
+    archive: ArchiveIcon,
+    music: MusicIcon,
+    text: TextIcon,
+    pdf: PdfIcon,
+    table: TableIcon,
+};
+
+export interface FilePreviewProps {
+    id?: string;
+
+    className?: string;
+    qa?: string;
+
+    file: File;
+    previewSrc?: string;
+    description?: string;
+
+    onClick?: MouseEventHandler<HTMLDivElement>;
+    actions?: FilePreviewActionProps[];
+}
+
+export const FilePreview: FC<FilePreviewProps> = ({
+    id: outerId,
+    className,
+    qa,
+    file,
+    previewSrc,
+    description,
+    onClick,
+    actions,
+}) => {
+    const defaultId = useUniqId();
+    const id = outerId ?? defaultId;
```

Comment:
outerId is not required, id is for component itself, there is no need or user to override `id`s required for a11y